lib/fizzy/leb128.hpp

```diff
@@ -19,11 +19,16 @@ std::pair<T, const uint8_t*> leb128u_decode(const uint8_t* input, const uint8_t*
         if (input == end)
             throw parser_error("Unexpected EOF");
 
-        // TODO this ignores the bits in the last byte other than the least significant one
-        // So would not reject some invalid encoding with those bits set.
         result |= static_cast<T>((static_cast<T>(*input) & 0x7F) << result_shift);
         if ((*input & 0x80) == 0)
+        {
+            constexpr auto all_ones = std::numeric_limits<T>::max();
+            const auto unused_bits_mask = ~static_cast<uint8_t>(all_ones >> result_shift);
+            if (*input & unused_bits_mask)
```

Comment:
```suggestion
            if (*input != (result << result_shift))
```

Reply:
yeah this works, but with the right shift instead :+1:

Reply:
Right